### DISCUSSION TITLE
Including variable that can be calculated at build time

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [vercel/next.js#13019] [Help] [answered]
If I have a value that can be calculated at build time, how do I make sure it is? Right now I'm doing the calculation just before a page function, like:
```js
let years = [2017, 2020, 2025, 1996];
years = years
  .filter((year, index) => years.indexOf(year) === index)
  .sort((yearA, yearB) => {
    return yearA < yearB ? 1 : -1;
  });

const Home = () => {
  return <>{years}</>
```
Is this the way I should do it? I know this doesn't need to go inside the page function, but where else should it go if it doesn't fit here?

**Comment by Timer** [comment, marked as answer]
This data would be better computed within `getStaticProps` so that the code could be eliminated.

https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation
